apps/studio/src/utils/sitemap.ts

```diff
@@ -0,0 +1,101 @@
+import type { IsomerSitemap } from "@opengovsg/isomer-components"
+import type { Resource } from "@prisma/client"
+
+type ResourceDto = Omit<
+  Resource,
+  "id" | "parentId" | "publishedVersionId" | "draftBlobId"
+> & {
+  id: string
+  parentId: string | null
+}
+
+const getSitemapTreeFromArray = (
+  resources: ResourceDto[],
+  parentId: string | null,
+  path: string,
+): IsomerSitemap[] | undefined => {
+  if (resources.length === 0) {
+    return undefined
+  }
+
+  const children = resources.filter((resource) => {
+    if (parentId === null) {
+      return resource.parentId === null && resource.permalink !== ""
```

Comment:
shud just use the type to compare cos that's the authoritative source!

Reply:
Updated in 74f51cb54a95ded6916da5469ee9dfd9e27e55ff.